swatch-producer-aws/src/main/java/com/redhat/swatch/aws/service/AwsBillableUsageAggregateConsumer.java

```diff
@@ -105,21 +110,38 @@ public void process(BillableUsageAggregate billableUsageAggregate) {
           billableUsageAggregate);
       return;
     } else {
-
       log.info("Processing billable usage message: {}", billableUsageAggregate);
     }
 
     AwsUsageContext context;
     try {
       context = lookupAwsUsageContext(billableUsageAggregate);
+    } catch (SubscriptionCanNotBeDeterminedException e) {
+      if (!isUsageDateValid(Clock.systemUTC(), billableUsageAggregate)) {
+        log.warn(
+            "Skipping billable usage with id={} orgId={} remittanceUUIDs={} because the subscription was not found and the snapshot '{}' is past the aws time window of '{}'",
+            billableUsageAggregate.getAggregateId(),
+            billableUsageAggregate.getAggregateKey().getOrgId(),
+            billableUsageAggregate.getRemittanceUuids(),
+            billableUsageAggregate.getWindowTimestamp(),
+            awsUsageWindow);
+        emitErrorStatusOnUsage(billableUsageAggregate, BillableUsage.ErrorCode.INACTIVE);
```

Comment:
Do we want to actually emit a status here? On the Azure side, we have a unit test, shouldNotSendMessageToDeadLetterQueueIfSnapshotDateIsOutOfTheTimeWindow that specifically checks these cases don't get retried.

Or perhaps, we emit the status but then filter on the billable-usage side in com.redhat.swatch.billable.usage.services.BillableUsageStatusConsumer#process to leave these out of the DLT?

@kflahert Any idea here if Azure's a different case from AWS?